### DISCUSSION TITLE
How to make VSCode prioritize the installed components over radix?

### DISCUSSION THREAD

**Discussion by benjick** [shadcn-ui/ui#2004] [Q&A] [answered]
Hello!

Every time is copy/paste a component and select "import all missing deps" VSCode prioritizes the Radix-imports over the components I have installed. Is there a way to fix this?

**Comment by yuki-yamamura** [comment, marked as answer]
If you use TypeScript, add the line below to `settings.json`. It will work after reloading your VSCode.

```json
  "typescript.preferences.autoImportFileExcludePatterns": ["@radix-ui"]
```

ref: https://github.com/shadcn-ui/ui/discussions/2535#discussioncomment-8226766

**Reply by benjick** [reply]
Thank you for this! 🙏
